src/Core/Logger.php

```diff
@@ -317,25 +317,37 @@ public static function debug($message, $context = [])
 		self::getApp()->saveTimestamp($stamp1, 'file');
 	}
 
-    /**
-     * @brief Logs the given message at the given log level
-     *
-     * @param string $msg
-     * @param string $level
+	/**
+	 * @brief Logs the given message at the given log level
+	 *
+	 * @param string $msg
+	 * @param string $level
 	 *
 	 * @throws \Exception
 	 * @deprecated since 2019.03 Use Logger::debug() Logger::info() , ... instead
-     */
-    public static function log($msg, $level = LogLevel::INFO)
-    {
+	 */
+	public static function log($msg, $level = LogLevel::INFO)
+	{
 		if (!isset(self::$logger)) {
 			return;
 		}
 
-        $stamp1 = microtime(true);
-		self::$logger->log($level, $msg);
-        self::getApp()->saveTimestamp($stamp1, "file");
-    }
+		$callers = debug_backtrace();
+
+		if (count($callers) > 1) {
+			$function = $callers[1]['function'];
```

Comment:
This function is deprecated and will be phased out as we start to use `Logger::debug`, `Logger::info`, `Logger::notice`, `Logger::warning` and `Logger::error`.

If you make a common function to prepare the context, don't forget to increment the callstack index number.

Reply:
The other functions are having ```$context``` as a parameter. How should that be handled?

Reply:
Append additional keys to the `$context` array or merge to get file, line and function first.

Reply:
the `$context` array is normally specific input based on the input for the current log entry.
For general additions to logging outputs, please add a new processor at
https://github.com/friendica/friendica/blob/a286bc98c651e6ec471c64bfa331e9453993ca4e/src/Util/LoggerFactory.php#L32

There's a common processor for adding `debug_backtrace` information here
https://github.com/Seldaek/monolog/blob/master/src/Monolog/Processor/IntrospectionProcessor.php

If there are too much information, we can use an own `FriendicaProcessor.php`, which I would add to `Util/Logger`, where we can define exactly which debug output we need. We can also define, at which log level the debug information should appear (currently it starts with `LogLevel::WARNING`)

Reply:
@nupplaphil I currently don't understand this whole new logger construct. Possibly you could create a PR where only the file, the line number and the function (without the class) is added at some appropriate place in the log output?

Reply:
I know. This is on my ToDo-List. I'm currently a little bit busy with fixing the API-tests, which will take another 1 or 2 week I guess.

Should I fix the fingerprint-issue instead? Is it more important?

Reply:
I would suggest using the function ```uniqid()``` for this.